components/script/dom/browsingcontext.rs

```diff
@@ -104,13 +114,13 @@ impl BrowsingContext {
     /// Change the Window that this browsing context's WindowProxy resolves to.
     // TODO: support setting the window proxy to a dummy value,
     // to handle the case when the active document is in another script thread.
-    pub fn set_window_proxy(&self, window: &Window) {
+    fn set_window_proxy<W: DomObject>(&self, window: &W, traps: &ProxyTraps) {
```

Comment:
Ditto. The problem was that when I started this, `XOriginWindow` was a global, but not a `GlobalScope`. Now it is, so I can make this a `GlobalScope`, so we can statically check that it's only called with globals.